unit_testing/test_register.py

```diff
@@ -167,11 +189,22 @@ def test_register_step_ants_registration(step2_data):
      verbose=cli_params.verbose,
     )
 
-@pytest.mark.skip(reason="TODO")
-def test_register_step_ants_slice_regularized_registration():
-     """
-     """
-     raise NotImplementedError()
+
+def test_register_step_ants_slice_regularized_registration(step_axial_data_in_same_space):
+    src, dest, step, cli_params = step_axial_data_in_same_space
+
+    warp_forward_out, warp_inverse_out = register_step_ants_slice_regularized_registration(
+        src=src,
+        dest=dest,
+        step=step,
+        metricSize='4')
+
+    # Verify integrity of the output Tx Ty file
+    txty_result = np.genfromtxt('step1TxTy_poly.csv', skip_header=1, delimiter=',')
+    txty_groundtruth = np.genfromtxt(
+        sct_test_path('mt', 'step1TxTy_poly_groundtruth.csv'), skip_header=1, delimiter=',')
+    assert all([a == pytest.approx(b, abs=1e-14) for a, b in zip(txty_result, txty_groundtruth)])
```

Comment:
Oh, careful here, switching to using numpy arrays here means that `==` is doing a vector comparison. Is that going to work right?


You should be able to say 

```suggestion
    assert txty_result == pytest.approx(txty_groundtruth, abs=1e-14)
```

or


```suggestion
    assert np.isclose(txty_result, txty_groundtruth, atol=1e-14).all()
```

Reply:
I guess `pytest.approx` understands numpy?

```
>>> A
array([[ 1.00439255, -0.15176018],
       [ 1.04645709, -0.621179  ],
       [ 0.90494934, -0.86215749],
       [ 0.78472878, -0.89309504],
       [ 0.8906549 , -0.732391  ]])

>>> A == pytest.approx(A+0.0000000000001, abs=1e-11)
True
>>> A == pytest.approx(A+0.0000000000001, abs=1e-12)
True
>>> A == pytest.approx(A+0.0000000000001, abs=1e-13)
False
>>> A == pytest.approx(A+0.0000000000001, abs=1e-14)
False
>>> A == pytest.approx(A+0.0000000000001, abs=1e-15)
False
```

Reply:
hum, the np.isclose does not work:
~~~
txty_result[0,1]
-0.151760183773
txty_groundtruth[0,1]
-0.151760183772956
np.isclose(txty_result, txty_groundtruth, atol=1e-20).all()
True
~~~

Reply:
whereas pytest.approx does the trick.
thank you, implemented in 6b66ae770ec85650a74f3b201152831f72d491f0